RUFAS/biophysical/manure/storage/solids_storage_calculator.py

```diff
@@ -0,0 +1,213 @@
+
+DEFAULT_LAG_TIME: int = 2
+"""Default lag time used in the calculation of the carbon decomposition rate (days). Default is set to 2."""
+
+DEFAULT_DAYS_SINCE_LAST_TURNING: int = 1
+"""Default days since the previous compost turning event (days). Default is set to 1."""
+
+COMPOSTING_DECOMPOSITION_TEMPERATURE: float = 60.0
+"""The temperature of the inner compost layer at which microbial growth and decomposition is maximized (C)."""
+
+DEFAULT_MOLE_FRACTION_OF_OXYGEN: float = 0.15
+"""The default mole fraction of oxygen in the air wihtin the compost layer."""
+
+OXYGEN_HALF_SATURATION_CONSTANT: float = 0.02
+"""The half saturation constant of Oxygen gas (O2)"""
+
+AMBIENT_AIR_MOLE_FRACTION_OF_OXYGEN: float = 0.21
+"""The mole fraction of oxygen in ambient air."""
+
+
+class SolidsStorageCalculator:
+    """
+    This class contains methods to calculate the carbon decomposition, methane emission,
+    nitrogen loss to leaching, and dry matter loss of the current day.
+    The calculations are based on the composting type and the manure temperature.
+    The methods are static and can be called without creating an instance of the class.
+    """
+
+    @staticmethod
+    def _calculate_nitrogen_loss_to_leaching(composting_type: CompostingType, received_manure_nitrogen: float) -> float:
+        """
+        This function calculates the amount of nitrogen leached out of the manure-bedding
+        pile of the current day.
+
+        Parameters
+        ----------
+        composting_type : CompostingType
+            The type of composting being used.
+        received_manure_nitrogen : float
+            The nitrogen content of the received manure, kg.
+
+        Returns
+        -------
+        float
+            The total nitrogen loss to leaching of the current day, kg.
+        """
+
+        return FRACTION_NITROGEN_LOST_TO_LEACHING[composting_type] * received_manure_nitrogen
+
+    @staticmethod
+    def _calculate_dry_matter_loss(methane_emission: float, carbon_decomposition: float) -> float:
+        """
+        This function calculates the total dry matter loss of the current day.
+
+        Parameters
+        ----------
+        methane_emission : float
+            The methane emission of the current day, kg/day.
+        carbon_decomposition : float
+            The carbon decomposition of the current day, kg/day.
+
+        Returns
+        -------
+        float
+            The total dry matter loss of the current day, kg/day.
+        """
+        return 2 * carbon_decomposition + methane_emission
+
+    @staticmethod
+    def _calculate_carbon_decomposition(
+        manure_temperature: float, non_degradable_volatile_solids: float, degradable_volatile_solids: float
+    ) -> float:
+        """
+        This function calculates the total carbon decomposition of the current day.
+
+        Parameters
+        ----------
+        manure_temperature : float
+            The manure temperature of the current day, Celsius.
```

Comment:
Two things here:
- I changed manure to "decomposing material" here; we can/should use this verbiage anywhere we are currently using either "compost" or "manure" in order to make docstrings generalized to compost vs. bedded pack manure vs. open lot manure. 
- Should we add a note here and in other applicable places where temperature is a parameter that temp is not always strictly ambient temperature? I did not include the formal constant name because it doesn't exist yet, I would suggest something like "DEFAULT_CARBON_DECOMP_TEMPERATURE"
- This is grammar police territory, but I think we want to use "**on** the current day" rather than "**of** the current day" unless this is a RuFaS-wide thing that I'm not aware of

```suggestion
            The temperature of decomposing material on the current day, Celsius. In Composting, this value is equal to the ambient temperature on the current day. In Open Lot and Compost Bedded Pack Barn, this value is set to a default/constant value (30 C).
```